src/main/java/org/adrian/com/spring/controlers/BinarySearchController.java

```diff
@@ -0,0 +1,27 @@
+package org.adrian.com.spring.controlers;
+
+import org.adrian.com.spring.services.BinarySearchService;
+import org.adrian.com.spring.models.AlgorithmsResult;
+import org.adrian.com.spring.models.BinarySearchRequest;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.*;
+
+@RestController
+@RequestMapping("/binarySearch")
+public class BinarySearchController {
+    @Autowired
+    BinarySearchService binarySearchService;
```

Comment:
złota zasada Sprina i jedno z częstszych pytań na rozmowach o pracę - rób dependency injection przez konstruktor.
Wystarczy że klasa jest annotowana przez spring, tak jak tu RestController, wtedy Binary search robisz private final i inicjujesz przez konstruktor. NIe potrzeba wtedy nawet adnotacji Autowire na konstruktorze, spring to ogarnie
Poza tym unikaj default scope dla pól